[Docs] Restructure of the Host List

BreezeQS opened the pull request:
### Type

- [ ] Bugfix
- [x] Enhancement
- [ ] New feature

### Description of the changes
Updates link in getting started page to reflect doc's new purpose.
Adds info about unsupported platforms.
Adds Lightsail, Hetzner, Contabo, Galaxygate and Ramnode to providers list. 
Rephrases description for hosting providers.
Replaces host list tables with text.
Re-arranges the location of selfhosting info.
Adds some details about the practice of hosting.

Sorry if i made a mistake, this is my first PR.

Commit:
Update hostlist

My first pr??

Commit:
Rephrase of unsupported warning

Exact reason for lack of container support added. Information about docker support added in case someone finds that repo and tries it.

Commit:
Remove azure from free list

Later found unreasonable to include it due to the 750 hours being spread across the entire year, ends up less than 24 hours a week.

Commit:
Oops

Fixed something that caused docs build to error

bobloy commented on docs/host-list.rst:
```diff
@@ -19,12 +19,13 @@ Hosting Information
+    conditions of being a supported platform; platforms such as Heroku, 
+    Pterodactyl, repl.it, Termux, LXC and alike are **NOT** officially supported. 
+    Docker support found in Cog-Creators GitHub is also a work in progress 
+    and not ready for daily use. Workarounds for getting Red  running on those 
```
```suggestion
    and not ready for daily use. Workarounds for getting Red running on those 
```
Extra space.

Commit:
Removal of extra space

Fix of spelling error noticed during review

Co-authored-by: bobloy <alboblexloy@gmail.com>

Commit:
Fix capitalization error

Idk if this was really necessary, but oh well

Commit:
Rephrase free tier note

Commit:
Rephrase docker warning

ghost commented:
This should be ready for another review

Commit:
Small changes

Added galaxygate and ramnode, removed contabo, rephrased a few things, added voice region tip.

Commit:
Small fix

Just realized i accidentailly removed Contabo

Flame442 commented on docs/host-list.rst:
```diff
@@ -2,90 +2,150 @@
 
-This is a list of the recommended VPS providers.
+
+| For your instance of Red to stay online it needs to be hosted in a system.
```
```suggestion
| For your instance of Red to stay online 24/7, it needs to be hosted on a dedicated system.
```

Flame442 commented on docs/host-list.rst:
```diff
@@ -2,90 +2,150 @@
+    Pterodactyl, repl.it, Termux, LXC and alike are **NOT** officially supported. 
+    Docker support found in GitHub is also a work in progress and not ready
+    for daily use. Workarounds for getting Red running on those platforms
+    are imperfect due to Red's nature. And one will not be able to recieve
```
```suggestion
    are imperfect due to Red's nature. You will not be able to receive
```

Flame442 commented on docs/host-list.rst:
```diff
@@ -2,90 +2,150 @@
+------------------------------------
+
+| One can host Red in a VPS running Linux or Windows. Using a Linux VPS is the
+  recommened option. Dedicated servers also work but are overpowered and cost 
```
```suggestion
  recommended option. Dedicated servers also work but are overpowered and cost 
```

Flame442 commented on docs/host-list.rst:
```diff
@@ -2,90 +2,150 @@
+| One can host Red in a VPS running Linux or Windows. Using a Linux VPS is the
+  recommened option. Dedicated servers also work but are overpowered and cost 
+  ineffective unless one plans to run a very large bot or use their server for 
+  more than than just hosting Red. After setup, Red can be moved to a different 
```
```suggestion
  more than just hosting Red. If you have already created an instance, Red can be moved to a different 
```

Flame442 commented on docs/host-list.rst:
```diff
@@ -2,90 +2,150 @@
+  starting point. One should conduct their own research and come to
+  a conclusion depending on their needs and budget, taking into account
+  providers not listed here if desired. The key is the provider offering 
+  a OS supported by Red.
```
```suggestion
  an OS supported by Red.
```

Flame442 commented on docs/host-list.rst:
```diff
@@ -2,90 +2,150 @@
+ "Droplet".
+
+| `OVH <https://us.ovhcloud.com/vps/>`_ is a company focused on providing hosting
+ and cloud services with locations in the Europe, North America and Asia Pacific.
```
```suggestion
 and cloud services with locations in Europe, North America and Asia Pacific.
```

Flame442 commented on docs/host-list.rst:
```diff
@@ -2,90 +2,150 @@
+ provider with locations in Germany and Finland.
+
+| `Contabo <https://contabo.com/>`_ is also a German VPS and dedicated server provider
+ with locations in Germany and United States.
```
```suggestion
 with locations in Germany and the United States.
```
I know, english is confusing sometimes.....

Flame442 commented on docs/host-list.rst:
```diff
@@ -2,90 +2,150 @@
 
-Other than that... no. There is no good free VPS hoster, outside of
-persuading somebody to host for you, which is incredibly unlikely.
+| Excluding the above, there is no recommended free VPS host. Persuation of
```
```suggestion
| Excluding the above, there is no recommended free VPS host. Persuasion of
```

Flame442 commented on docs/host-list.rst:
```diff
@@ -2,90 +2,150 @@
+Hosting on a VPS or Dedicated Server
+------------------------------------
+
+| One can host Red in a VPS running Linux or Windows. Using a Linux VPS is the
```
```suggestion
| You can host Red in a VPS running Linux or Windows. Using a Linux VPS is the
```
For ""formal"" writing, you don't want to use "you", but for documentation it's fine

BreezeQS replied:
I just tend to use "One" more frequently than most, thank you!

Flame442 commented on docs/host-list.rst:
```diff
@@ -2,90 +2,150 @@
 
-You can always self-host on your own hardware.
-A Raspberry Pi 3 will be more than sufficient for small to medium sized bots.
+| It's possible to self host Red with one's own hardware. A Raspberry Pi 3 
```
```suggestion
| It's possible to self host Red with your own hardware. A Raspberry Pi 3 
```

Flame442 commented on docs/host-list.rst:
```diff
@@ -2,90 +2,150 @@
-A Raspberry Pi 3 will be more than sufficient for small to medium sized bots.
+| It's possible to self host Red with one's own hardware. A Raspberry Pi 3 
+  will have enough computing capacity to handle a small to medium sized bot. 
+  One also has the option of building a computer for this purpose or purchasing 
+  a  rack server as any modern hardware should work without issues. But keep in
+  mind this option can leaves one vulnerable to factors outside their control
+  such as power outages.
```
```suggestion
  You can also host on your own computer or rack server. Any modern hardware 
  should work without issues. However, this option leaves you responsible for
  keeping the bot online by paying for electricity costs and dealing with power outages.
```
I rewrote this section a bit, changed the focus to what I think is the major drawbacks to selfhosting. If you want to rewrite it again in a different way, feel free.

BreezeQS replied:
Better than anything i could come up with

Flame442 commented on docs/host-list.rst:
```diff
@@ -2,90 +2,150 @@
+
+| The following are some common providers suitable for hosting Red. With
+  each having their pros and cons, this list is mainly intended to act as a
+  starting point. One should conduct their own research and come to
+  a conclusion depending on their needs and budget, taking into account
```
```suggestion
  starting point. You should conduct your own research and come to
  a conclusion depending on your needs and budget, taking into account
```

Flame442 reviewed (changes requested):
Mostly some grammar changes. Thank you for your contribution!

Commit:
Update docs/host-list.rst

Co-authored-by: Flame442 <34169552+Flame442@users.noreply.github.com>

Commit:
Update docs/host-list.rst

Co-authored-by: Flame442 <34169552+Flame442@users.noreply.github.com>

Commit:
Update docs/host-list.rst

Co-authored-by: Flame442 <34169552+Flame442@users.noreply.github.com>

Commit:
Update docs/host-list.rst

Co-authored-by: Flame442 <34169552+Flame442@users.noreply.github.com>

Commit:
Update docs/host-list.rst

Co-authored-by: Flame442 <34169552+Flame442@users.noreply.github.com>

Commit:
Update docs/host-list.rst

Co-authored-by: Flame442 <34169552+Flame442@users.noreply.github.com>

Commit:
Update docs/host-list.rst

Co-authored-by: Flame442 <34169552+Flame442@users.noreply.github.com>

Commit:
Update docs/host-list.rst

Co-authored-by: Flame442 <34169552+Flame442@users.noreply.github.com>

Commit:
Update docs/host-list.rst

Co-authored-by: Flame442 <34169552+Flame442@users.noreply.github.com>

Commit:
Update docs/host-list.rst

Co-authored-by: Flame442 <34169552+Flame442@users.noreply.github.com>

Commit:
Update docs/host-list.rst

Co-authored-by: Flame442 <34169552+Flame442@users.noreply.github.com>

Commit:
Update docs/host-list.rst

Co-authored-by: Flame442 <34169552+Flame442@users.noreply.github.com>

Flame442 reviewed (approved):
All set! Thanks again!

ghost commented:
Thanks!!
